azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/http/http_transport.py

```diff
@@ -0,0 +1,149 @@
+        if callback:
+            callback()
+
+    def send_event(self, message, callback=None):
+        path = "/devices/" + self._auth_provider.device_id
+        if self._auth_provider.module_id is not None:
+            path += "/modules/" + self._auth_provider.module_id
+        path += "/messages/events/"
+
+        hostname = None
+        if hasattr(self._auth_provider, "gateway_hostname"):
+            hostname = self._auth_provider.gateway_hostname
+        if not hostname or len(hostname) == 0:
+            hostname = self._auth_provider.hostname
+
+        request_url = "https://" + hostname + path
+        request_params = {"api-version": "2018-06-30"}
+
+        logger.info("HTTPTransport: send_event: %s", path)
+
+        message_to_send = message
+
+        if not isinstance(message, Message):
+            message_to_send = Message(message)
+
+        headers = self._headers_from_message(message_to_send)
+
+        # TODO: user agent
+        # TODO: verify with the x509_auth branch
+        client_cert = None
+        if hasattr(self._auth_provider, "cert"):
+            client_cert = (self._auth_provider.cert, self._auth_provider.key)
+        else:
+            headers["Authorization"] = self._auth_provider.get_current_sas_token()
+
+        try:
+            if hasattr(self._auth_provider, "ca_cert") and self._auth_provider.ca_cert is not None:
+                ca_cert = self._auth_provider.ca_cert
+                resp = requests.post(
+                    request_url,
+                    params=request_params,
+                    data=message_to_send.data,
+                    headers=headers,
+                    verify=ca_cert,
+                )
+            else:
+                if client_cert:
+                    resp = requests.post(
+                        request_url,
+                        params=request_params,
+                        data=message_to_send.data,
+                        headers=headers,
+                        cert=client_cert,
+                    )
+                else:
+                    resp = requests.post(
+                        request_url,
+                        params=request_params,
+                        data=message_to_send.data,
+                        headers=headers,
+                    )
+            resp.raise_for_status()
+            logger.info("HTTP request was successful. status code: " + str(resp.status_code))
+            callback()
+        except requests.exceptions.SSLError as ex:
+            raise ex
+        except requests.exceptions.ConnectionError as ex:
+            raise ex
+        except requests.exceptions.HTTPError as ex:
+            logger.error("HTTP request failed with status code: " + str(resp.status_code))
+            # Probably need custom exception handling here to transport transport-specific exception into transport-agnostic exception?
+            # Do transports throw or callback with errors?
+            raise ex
```

Comment:
should there be a default except block `requests.exceptions.RequestException` for all other exceptions that request raises ?